### DISCUSSION TITLE
Create React App setup.ts

### DISCUSSION THREAD

**Discussion by rbjamaca** [decentralized-identity/veramo#411] [Q&A] [answered]
Hello, when following the guide on [Create React App](https://veramo.io/docs/guides/browser), I encountered an error when creating the setup.ts file:
`Type '(did: string, parsed: ParsedDID) => Promise<DIDDocument | null>' is not assignable to type 'DIDResolver'.
  Type 'Promise<DIDDocument | null>' is not assignable to type 'Promise<DIDResolutionResult>'.
    Type 'DIDDocument | null' is not assignable to type 'DIDResolutionResult'.
      Type 'null' is not assignable to type 'DIDResolutionResult'.`

![image](https://user-images.githubusercontent.com/58159983/110259750-50cfd180-7fe4-11eb-9029-5b570569c629.png)

Tried setting `strictNullChecks` to false but it still returns an error on start:
`Type '(did: string, parsed: ParsedDID) => Promise<DIDDocument>' is not assignable to type 'DIDResolver'.
  Type 'Promise<DIDDocument>' is not assignable to type 'Promise<DIDResolutionResult>'.
    Type 'DIDDocument' is missing the following properties from type 'DIDResolutionResult': didResolutionMetadata, didDocument, didDocumentMetadata  TS2322 `

**Comment by mirceanis** [comment, marked as answer]
I believe this is caused by a mismatched version of the  `did-resolver` library.
That library has a breaking change update in the latest version and veramo is not yet updated to support it.

Please downgrade your version of that library
